src/main/java/com/example/bookstore/repository/BookRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.example.bookstore.repository;
+
+import com.example.bookstore.model.Book;
+import java.util.List;
+
+public interface BookRepository {
+    Book save(Book book);
+
+    List<Book> findAll();
+
+    Book getBookById(Long id);
```

Comment:
better return optional here, not being able to find by id is an expected situation